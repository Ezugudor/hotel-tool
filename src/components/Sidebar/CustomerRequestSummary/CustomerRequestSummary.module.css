.Container{
  
}

.RequestInfoHead{
    
}

.RequestInfoCont{
    border-bottom:1px solid #956cc4;
    padding:10px
}

.RequestInfo{
    display:flex;
    justify-content:space-between;
    margin-bottom:5px;
}

.RILabel{

}

.RIValue{
    font-weight:bold;
    color:#fff;
    font-size:20px
}

.RIValueCount{
    color:#0ee89c
}