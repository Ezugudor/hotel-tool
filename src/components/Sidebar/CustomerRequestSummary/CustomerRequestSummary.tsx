import React, { useContext } from "react";
import { CustomerContext } from "../../../context";
import { countRequests, sumRequests } from "../../../helpers/Helpers";
import { RoomCategory } from "../../../models/RoomCategory";
import styles from "./CustomerRequestSummary.module.css"

export const CustomerRequestSummary: React.FunctionComponent = () => {

    const customerBids = useContext(CustomerContext);


    return (
        <div className={styles.Container}>
            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Total Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids)} </span>
                    </div>
                </div>
            </div>

            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Premium Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids, RoomCategory.PREMIUM)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids, RoomCategory.PREMIUM)} </span>
                    </div>
                </div>
            </div>

            <div className={styles.RequestInfoCont}>
                <h4 className={styles.RequestInfoHead}>Non-Premium Requests</h4>
                <div className={styles.RequestInfoBody}>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Count: </span>
                        <span className={`${styles.RIValue} ${styles.RIValueCount}`}> {countRequests(customerBids, RoomCategory.NON_PREMIUM)} </span>
                    </div>
                    <div className={styles.RequestInfo}>
                        <span className={styles.RILabel}>Amount: </span>
                        <span className={styles.RIValue}> € {sumRequests(customerBids, RoomCategory.NON_PREMIUM)} </span>
                    </div>
                </div>
            </div>
        </div>

    )
}