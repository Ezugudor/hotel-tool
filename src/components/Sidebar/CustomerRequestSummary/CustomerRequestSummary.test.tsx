import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CustomerRequestSummary } from './CustomerRequestSummary';



let customerRequestSummaryCont: ShallowWrapper;

beforeAll(() => {
    customerRequestSummaryCont = shallow(<CustomerRequestSummary />);
});


describe("CustomerRequestSummary Component Test", () => {
    it('should match with existing snapshot', () => {
        expect(customerRequestSummaryCont.html()).toMatchSnapshot();
    })


})
