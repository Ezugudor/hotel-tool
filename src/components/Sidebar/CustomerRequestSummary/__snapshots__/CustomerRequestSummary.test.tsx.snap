// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerRequestSummary Component Test should match with existing snapshot 1`] = `"<div><div><span class=\\"label\\">Total Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Total Requests Amount: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Premium Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Premium Requests Amount: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Non Premium Requests Count: </span><span class=\\"value\\"> 0 </span></div><div><span class=\\"label\\">Non Premium Requests Amount: </span><span class=\\"value\\"> 0 </span></div></div>"`;
